Currency Exchange]: Improvements (#2601)
exercism/python: exercises/concept/currency-exchange/.docs/instructions.md

## 3. Calculate value of bills

Create the `get_value_of_bills()` function, taking 2 parameters:

1. `denomination` : The value of a single bill.
2. `number_of_bills` : Amount of bills you received.

This function should return the total value of the given *bills*.

```python
>>> get_value_of_bills(5, 128)
640
```

## 4. Calculate number of bills

Create the `get_number_of_bills()` function, taking `budget` and `denomination`.

> This function should return the *number of bills* that you can get using the *budget*.

Reviewer: ```suggestion
This function should return the *number of bills* that you can get using the *budget*.

**Note: ** You can only receive _whole bills_, not fractions of bills,  so remember to divide accordingly.
```

> 

Reviewer: ```suggestion
**Note: ** You can only receive _whole bills_, not fractions of bills,  so remember to divide accordingly.
```

## 5. Calculate value after exchange

Create the `exchangeable_value()` function, taking `budget`, `exchange_rate`, `spread`, and `denomination`.

> Parameter `spread` is the *percentage taken* as exchange fee. If `1.00 EUR == 1.20 USD` and the *spread* is `10%`, the actual exchange will be: `1.00 EUR == 1.32 USD`.

Reviewer: ```suggestion
Parameter `spread` is the *percentage taken* as an exchange fee.
If `1.00 EUR == 1.20 USD` and the *spread* is `10%`, the actual exchange will be: `1.00 EUR == 1.32 USD`.
```

Reviewer 2: When I read `10%` I think `0.10`. It would be nice to clarify that `10%` is conveyed as `10`.

Author: I think it's good that we do not tell them, it'll give them something to find out... We cant hand them everything :)

> This function should return the maximum available value after *exchange rate* and the *denomination*.

Reviewer: ```suggestion
This function should return the maximum value of the new currency after calculating the *exchange rate* plus the *spread*.
Remember that the currency *denomination* is a whole number, and cannot be sub-divided.
```

> 95

Reviewer: ```suggestion
80
```

Reviewer: If we take the exemplar code :

```python
def exchangeable_value(budget, exchange_rate, spread, denomination):
    """

    :param budget: float - the amount of your money you are planning to exchange.
    :param exchange_rate: float - the unit value of the foreign currency.
    :param spread: int - percentage that is taken as an exchange fee.
    :param denomination: int - the value of a single bill.
    :return: int - maximum value you can get.
    """

    exchange_fee = (exchange_rate / 100) * spread
    actual_rate = exchange_rate + exchange_fee
    exchangeable_amount = int((budget / actual_rate) / denomination)
    return exchangeable_amount * denomination
```

and plug in the example `exchangeable_value(127.25, 1.20, 10, 20)` , the result is `80`, not `95`.

Author: ```python
exchangeable_value(127.25, 1.20, 10, 5)
95
```
```python
exchangeable_value(127.25, 1.20, 10, 20)
80
```

I ran the tests myself as well, these two are correct on my machine. It is true that `exchangeable_value(127.25, 1.20, 10, 20)` is not `95`, but I think you might've grabbed the wrong call, because in the instructions it says `exchangeable_value(127.25, 1.20, 10, 5)` == `95`. Or am I seeing this wrong myself?

Reviewer: No -- you are correct, @J08K  -- the spacing in the review window fooled me, and I should have viewed the whole file.  The `80` is indeed in the correct place.  😊
